redwood-treehouse/src/commonMain/kotlin/app/cash/redwood/treehouse/ZiplineTreehouseUi.kt

```diff
@@ -46,4 +49,12 @@ public interface ZiplineTreehouseUi : ZiplineService, EventSink {
   )
 
   public fun snapshotState(): StateSnapshot? = null
+
+  /** Access to the host that presents this UI. */
+  public interface Host : ZiplineService, ChangesSinkService {
+    public val uiConfigurations: StateFlow<UiConfiguration>
+    public val stateSnapshot: StateSnapshot?
+
+    public fun addOnBackPressedCallback(callback: OnBackPressedCallbackService): CancellableService
```

Comment:
Can we pass the `OnBackPressedDispatcherService` here? The fact that `OnBackPressedDispatcherService` has only one function is happenstance, and if we were to add more functions to `OnBackPressedDispatcherService`, I'd prefer to not have to also update `Host`.

Reply:
Great question!

The Host interface is a Redwood-internal implementation detail so we can reduce the number of ZiplineService instances we need to bridge, and also the number we need to close.

This is one of those things where we’ve gradually added more things as we add more capabilities, and we’re overdue to flatten them all both for forwards-compatibility and for performance.

Rather than having one ZiplineService-style adapter for every feature we bridge (changes sink, save, back, and soon exceptions), this flattens them all into a single interface to reduce the runtime setup. When we need a new capability, we don’t need a new overload and a new catch clause to fall back on.

Instead of adding new BackHandler features to BackHandlerService to bridge them, we will instead add them to this Host interface. It doesn’t limit our ability to introduce new features for back handling in the future.

Reply:
That makes sense. Thanks for clarifying!